ipaclient/install/client.py

```diff
@@ -2472,12 +2486,9 @@ def sync_time(options, fstore, statestore):
 
     if not options.ntp_servers and not options.ntp_pool:
         ds = discovery.IPADiscovery()
-        ntp_servers = ds.ipadns_search_srv(cli_domain, '_ntp._udp',
-                                           None, break_on_first=False)
-        if not ntp_servers and not options.unattended:
-            options.ntp_servers, options.ntp_pool = timeconf.get_time_source()
-        else:
-            options.ntp_servers = ntp_servers
+        options.ntp_servers = ds.ipadns_search_srv(
```

Comment:
Is it bad form to update the options passed in vs those things detected? Could this make debugging harder later?

Reply:
Hello @rcritten, 

yes I agree, actual behaviour could make debugging harder however i think approach described below is good and I will add another logging message and comments to the code itself.
```python
if installation is interactive and no ntp options had been passed:
    ask user for ntp options interactively
if no ntp options are passesd to installation:
    try autodiscovery for DNS record
if installation has no ntp options:
    skip configuration and use default one

synchronize time
```
Would that be acceptable alongside with update of design document?

Reply:
I agree with the steps and the purpose. I found it quite jarring to be prompted for NTP after answering yes to proceed with the installation, so I'm glad you're addressing this!
Is adding a new set of variables, ntp_server/ntp_pool so bad? It seems wrong to twiddle with options.*. We do this other places in the code as well, e.g. hostname, cli_domain, cli_server,...

Reply:
Sorry I have misunderstood. Updating PR.

Reply:
https://github.com/freeipa/freeipa/blob/254ba5240e897da04eea06b4f02dea579a9a6c54/ipaclient/install/client.py#L2476